actions/mouths-of-watersheds/mouths-of-watersheds.py

```diff
@@ -0,0 +1,52 @@
+import argparse
+import csv
+from netCDF4 import Dataset
+from shapely.geometry import shape
+import geojson
+from mow.helpers import *
+from mow.geo_data_grid_2d.vic import VicDataGrid
+
+parser = argparse.ArgumentParser('Output a CSV file with the lon lat of the mouth of each watershed')
+parser.add_argument('nc', help='an RVIC-netCDF file')
+parser.add_argument('geojson', help='a geoserver output file describing the watersheds')
+parser.add_argument('csv', help='the csv file to output the results to')
+args = parser.parse_args()
+nc = Dataset(args.nc, "r")
+flow_direction = VicDataGrid.from_nc_dataset(nc, "flow_direction")
+grid = VIC_direction_matrix(flow_direction.lat_step, flow_direction.lon_step) 
+
+with open(args.geojson) as f:
+    with open(args.csv, 'w', encoding='UTF8') as x:
+        writer = csv.writer(x)
+        header = ['lon', 'lat', 'properties']
+        writer.writerow(header)
+        gj = geojson.load(f)
+        for i in gj['features']:
+            visited = set()
+            center = shape(i.geometry).centroid
+            try:
+                xy = flow_direction.lonlat_to_xy(center.coords[0])
+            except:
+                continue
```

Comment:
Suggest we emit a warning message here.

Reply:
My problem with emitting a warning message here is that basically this is just checking that a center is in the netcdf file. It is expected that the majority of watersheds will not be in the file. This would lead to a lot of error messages that don't really relate to the user

Reply:
> This would lead to a lot of error messages

This would lead to a lot of ~~error~~ *warning* messages.

Fixed that for you :smile:  Warning messages are easy to filter or easy(-ish) for the user to configure so that they only see the same warning once. In general, it's better to be verbose and let the user decide what they do, or do not need to know.

Reply:
How do I emit a warning message?

Reply:
Generally, `warnings.warn()`, but read the [full docs on the `warnings` module](https://docs.python.org/3/library/warnings.html) for some idea of the customization behaviour available.